### DISCUSSION TITLE
TypeScript Error on Build

### DISCUSSION THREAD

**Discussion by Rounak-stha** [vercel/next.js#50447] [Help] [answered]
### Summary

I am progressively migrating to `Nextjs 13` and with `typescript`. 
Whenever nextjs discovers a typescript file it automatically configures `ts` for us and I used that but during build, I am getting this error.

![image](https://github.com/vercel/next.js/assets/66557682/e949b68e-ae77-427c-839a-4e22997bc6bd)

**Comment by icyJoseph** [comment, marked as answer]
Well it is saying that typescript is not installed as a dependency in your project. 

If I remember correctly, when Next.js discovers a TS file, it tries to install typescript and other required dependencies. Did you commit that to your repository?

I can imagine that during build time on Vercel, they probably don't want to install packages, they respect/honour your package.json and lock file, for security reasons.

So when the build step tries to install something, it exists with code 1.

Why not just install TS and friends and commit that to the repo?

## Update

This is expected behavior.

If the build environment is a CI environment, then the build doesn't install anything, and it crashes out instead.

Locally I did:

```console
CI=true yarn build
yarn run v1.22.19
$ next build
- warn No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

It looks like you're trying to use TypeScript but do not have the required package(s) installed.

Please install typescript, @types/react, and @types/node by running:

	yarn add --dev typescript @types/react @types/node

If you are not trying to use TypeScript, please remove the tsconfig.json file from your package root (and any TypeScript files in your pages directory).

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by Rounak-stha** [comment]
I assumed that since `ts` and `tsx` files are automatically sensed and ts support is added, I didn't have to do anything else. The automatic generation of the `tsconfig` file gave me more confidence but it turns out `typescript` was not added as a dev-dependency.  Maybe it got accidentally removed during the development but I am over that error now. **Thanks**
